src/main/java/org/passay/CharacterRule.java

```diff
@@ -48,7 +48,7 @@ public CharacterRule(final CharacterData data, final int num)
    *
    * @param  n  number of characters to require where n &gt; 0
    */
-  public void setNumberOfCharacters(final int n)
+  private void setNumberOfCharacters(final int n)
```

Comment:
My thought pattern behind this was that since we have the setter under the constructor, and it is public yet not final, extending this class might yield to unexpected behavior.  This was also the reason why I declared some public Rules as final, solely due to having a non final public method under the constructor.